includes/user-functions.php

```diff
@@ -483,12 +483,14 @@ function edd_add_past_purchases_to_new_user( $user_id ) {
 		edd_send_user_verification_email( $user_id );
 
 		foreach( $payments as $payment ) {
-			if( intval( edd_get_payment_user_id( $payment->ID ) ) > 0 ) {
-				continue; // This payment already associated with an account
-			}
+			if ( is_object( $payment ) && $payment instanceof EDD_Payment ) {
+				if ( intval( edd_get_payment_user_id( $payment->ID ) ) > 0 ) {
```

Comment:
Since we have a known `EDD_Payment` object, can we just use `$payment->user_id`? `edd_get_payment_user_id()` does this, which means we're kicking up two instances of the same payment.

Reply:
Yeah we should use `$payment->user_id`.